### ISSUE TITLE
Reorganize the file structure?

### ISSUE THREAD

**Issue by PsiACE** [datafuselabs/datafuse#1367]
This draft focuses on the following points:

1. reorganizing the file structure.
2. unifying the name style `fuse` -> `datafuse`.

**Examples 1:**

- before:
 
```
├── fusestore
│   ├── README.md
│   └── store
│       ├── build.rs
│       ├── Cargo.toml
│       ├── proto
│       └── src
```

- after:

```
   └── store
       ├── README.md
       ├── build.rs
       ├── Cargo.toml
       ├── proto
       └── src
```

**Examples 2:**

- before:

```
[package]
name = "fuse-query"
```

- after:

```
[package]
name = "datafuse-query"
```

**Comment by PsiACE** [comment]
also, if we have plans to implement a full SQL dialect parser ourselves, then we can start by moving the sql module out of query into a separate component.

**Comment by zhang2014** [comment]
> reorganizing the file structure.
> unifying the name style fuse -> datafuse.

Look good to me.

**Comment by ZhiHanZ** [comment]
Does package names change affect binary file names?

**Comment by PsiACE** [comment]
> Does package names change affect binary file names?

Yes. I will modify the scripts and make the ci work correctly.

**Comment by BohuTANG** [comment]
LGTM :+1:

**Comment by ZhiHanZ** [comment]
> > Does package names change affect binary file names?
> 
> Yes. I will modify the scripts and make the ci work correctly.

pay special attentions to docker build part, and test infra needs to change as well, as an alternative method, we could rename package name for docker part

**Comment by PsiACE** [comment]
Closed by #1404 .
